packages/runtime/container-runtime/src/opLifecycle/batchManager.ts

```diff
@@ -169,14 +174,22 @@ export class BatchManager {
 		const startPoint = this.pendingBatch.length;
 		return {
 			rollback: (process: (message: BatchMessage) => void) => {
-				for (let i = this.pendingBatch.length; i > startPoint; ) {
+				const rollbackOps = this.pendingBatch.splice(startPoint);
+				for (let i = rollbackOps.length; i > 0; ) {
```

Comment:
Option for a cleaner look
```suggestion
				const rollbackOps = this.pendingBatch.splice(startPoint).reverse();
				for (const rollbackOp of rollbackOps) {
```